exorcist/taskdb.py

```diff
@@ -2,10 +2,13 @@
 import sqlalchemy as sqla
 import networkx as nx
 from .models import TaskStatus
+from datetime import datetime
 
-# remaining imports are for typing
-from typing import Optional, Iterable
+# imports for typing
+from typing import Optional, Iterable, Union
 from os import PathLike
+from sqlalchemy.sql.roles import StatementRole as SQLStatement
```

Comment:
another reason for splitting out the abc, now the abc depends on sqlalchemy thanks to this import